### ISSUE TITLE
Loadbalancer trying to make LEARNER a LEADER

### ISSUE THREAD

**Issue by rao-vasireddy** [YugaByte/yugabyte-db#1587]
tablet status on the current tablet leader tserver:

redis | **08ec006970774140aa55a97d02ee4fd2** | hash_split: [10912, 11594) | RUNNING | 1.09G | LEARNER: 172.151.18.27FOLLOWER: 172.151.23.152LEADER: 172.151.31.215
-- | -- | -- | -- | -- | --

tablet status on the new tserver where master is trying to move the leader to:

redis | **08ec006970774140aa55a97d02ee4fd2** | hash_split: [10912, 11594) | TABLET_DATA_COPYING | 0B
-- | -- | -- | -- | --

master leader log:

I0620 20:57:54.388131 28627 catalog_manager_util.cc:49] Load standard deviation is 28.2843 for 3 tservers in placement aws:us-west-2:us-west-2a for placement uuid 66bb346e-ded9-4399-bff0-ed5fb1c6dc62
I0620 20:57:55.337888 16800 cluster_balance.cc:803] Moving leader of **08ec006970774140aa55a97d02ee4fd2** from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:57:55.337965 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:57:55.337975 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet **08ec006970774140aa55a97d02ee4fd2**
I0620 20:57:55.338500 25901 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:57:56.338688 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:57:56.338748 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:57:56.338754 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2
I0620 20:57:56.339342 25902 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:57:57.339494 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:57:57.339547 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:57:57.339555 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2
I0620 20:57:57.340092 25903 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:57:58.340221 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:57:58.340270 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:57:58.340276 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2
I0620 20:57:58.340842 25904 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:57:59.340999 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:57:59.341060 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:57:59.341066 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2
I0620 20:57:59.341588 25905 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:58:00.341708 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:58:00.341802 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:58:00.341812 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2
I0620 20:58:00.342312 25907 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:58:01.342558 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:58:01.342612 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:58:01.342618 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2
I0620 20:58:01.343194 25935 async_rpc_tasks.cc:912] Leader step down done attempt=1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_uuid=308174a3909a4a8686f0e25fcf909922, error=code: LEADER_NOT_READY_TO_STEP_DOWN status { code: ILLEGAL_STATE message: "Leader not ready to step down as there are 1 active config peers in transition, 1 in committed. Configs:\nactive=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }\ncommit=opid_index: 44861885 peers { permanent_uuid: \"95267ae737194dd5a0557bc58202b88e\" member_type: VOTER last_known_private_addr { host: \"172.151.23.152\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"308174a3909a4a8686f0e25fcf909922\" member_type: VOTER last_known_private_addr { host: \"172.151.31.215\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } } peers { permanent_uuid: \"835605b3e1734b27837932c11ca2f6c8\" member_type: PRE_VOTER last_known_private_addr { host: \"172.151.18.27\" port: 9100 } cloud_info { placement_cloud: \"aws\" placement_region: \"us-west-2\" placement_zone: \"us-west-2a\" } }" source_file: "../../src/yb/consensus/raft_consensus.cc" source_line: 645 }, failed=1, should_remove=0 for tablet 08ec006970774140aa55a97d02ee4fd2.
I0620 20:58:02.343331 16800 cluster_balance.cc:803] Moving leader of 08ec006970774140aa55a97d02ee4fd2 from TS 308174a3909a4a8686f0e25fcf909922 to 835605b3e1734b27837932c11ca2f6c8
I0620 20:58:02.343400 16800 async_rpc_tasks.cc:863] Prep Leader step down 1, leader_uuid=308174a3909a4a8686f0e25fcf909922, change_ts_uuid=308174a3909a4a8686f0e25fcf909922
I0620 20:58:02.343415 16800 async_rpc_tasks.cc:892] Stepping down leader 308174a3909a4a8686f0e25fcf909922 for tablet 08ec006970774140aa55a97d02ee4fd2

**Comment by rao-vasireddy** [comment]
I also noticed that when a tserver is stopped, the leaders on that node are moved to the other nodes even if some of the nodes have not finished bootstrapping the tablets for which the leader was moved there

**Comment by rahuldesirazu** [comment]
Was digging into LB code, we only consider running tablet peers for leader moves but peers are only marked as running at the end of OpenTablet, so its strange behavior that leader moves were being considered for bootstrapping peers. 

In UpdateTablet called when creating in memory state:
```
    const tablet::RaftGroupStatePB& tablet_state = replica.second.state;
      if (tablet_state == tablet::RUNNING) {
        ts_meta_it->second.running_tablets.insert(tablet_id);
        ++tablet_meta.running;
        ++total_running_;
      } else if (tablet_state == tablet::BOOTSTRAPPING || tablet_state == tablet::NOT_STARTED) {
        // Keep track of transitioning state (not running, but not in a stopped or failed state).
        ts_meta_it->second.starting_tablets.insert(tablet_id);
        ++tablet_meta.starting;
        ++total_starting_;
      }
```

In HandleLeaderMoves:
```
  const set<TabletId>& leaders = state_->per_ts_meta_[high_load_uuid].leaders;
  const set<TabletId>& peers = state_->per_ts_meta_[low_load_uuid].running_tablets;
```

Note that we only add to the running tablets set if the state is RUNNING, and leader moves only considers peers that have running tablets.

**Comment by bmatican** [comment]
I wonder if the fix @hectorgcr is working on for #1686 would address this as well?
